crates/cli/src/subcommands/version.rs

```diff
@@ -1,33 +1,82 @@
-use clap::{Arg, ArgAction::SetTrue, ArgMatches};
+use std::ffi::OsString;
+use std::path::PathBuf;
+use std::process::{Command, ExitCode};
 
-pub const CLI_VERSION: &str = env!("CARGO_PKG_VERSION");
-
-use crate::config::Config;
+use anyhow::Context;
+use clap::{ArgMatches, Args};
+use spacetimedb_paths::cli::BinFile;
+use spacetimedb_paths::{FromPathUnchecked, RootDir, SpacetimePaths};
 
 pub fn cli() -> clap::Command {
-    clap::Command::new("version")
-        .about("Print the version of the command line tool")
-        .after_help("Run `spacetime help version` for more detailed information.\n")
-        .arg(
-            Arg::new("cli")
-                .long("cli")
-                .action(SetTrue)
-                .help("Prints only the CLI version"),
-        )
+    Version::augment_args(clap::Command::new("version"))
+}
+
+/// Manage installed spacetime versions
+///
+/// Run `spacetime version --help` to see all options.
+#[derive(clap::Args)]
+#[command(disable_help_flag = true)]
+struct Version {
+    /// The args to pass to spacetimedb-update
+    #[arg(allow_hyphen_values = true, num_args = 0..)]
+    args: Vec<OsString>,
 }
 
-pub async fn exec(_config: Config, args: &ArgMatches) -> Result<(), anyhow::Error> {
-    if args.get_flag("cli") {
-        println!("{}", CLI_VERSION);
-        return Ok(());
+pub async fn exec(paths: &SpacetimePaths, root_dir: Option<&RootDir>, args: &ArgMatches) -> anyhow::Result<ExitCode> {
+    let args = args.get_many::<OsString>("args").unwrap_or_default();
+    let bin_path;
+    let bin_path = if let Some(artifact_dir) = running_from_target_dir() {
+        let update_path = artifact_dir
+            .join("spacetimedb-update")
+            .with_extension(std::env::consts::EXE_EXTENSION);
+        anyhow::ensure!(
+            update_path.exists(),
+            "running `spacetime version` from a target/ directory, but the spacetimedb-update
+             binary doesn't exist. try running `cargo build -p spacetimedb-update`"
+        );
+        bin_path = BinFile::from_path_unchecked(update_path);
+        &bin_path
+    } else {
+        &paths.cli_bin_file
+    };
+    let mut cmd = Command::new(bin_path);
+    if let Some(root_dir) = root_dir {
+        cmd.arg("--root-dir").arg(root_dir);
     }
+    cmd.arg("version").args(args);
+    let applet = "spacetimedb-update";
+    #[cfg(unix)]
+    let result = {
+        use std::os::unix::process::CommandExt;
+        cmd.arg0(applet);
+        let err = cmd.exec();
+        Err::<std::process::ExitStatus, _>(err)
+    };
+    #[cfg(windows)]
+    let result = {
+        cmd.env("SPACETIMEDB_UPDATE_MULTICALL_APPLET", applet);
```

Comment:
is this needed because the `args[0]` behavior is different on windows than on linux e.g. if going via a shortcut/symlink?

What's the advantage of doing this differently on windows vs linux than just always doing this env-based approach?

Reply:
Ultimately this is a hack to get behavior like argv0 on windows. In the future, there'll hopefully be a function akin to argv0 on windows, but for now this is the best approximation. https://github.com/rust-lang/rust/issues/66510

Reply:
That makes sense. I think my question remains though: if the "hack" is good enough for windows, why not just use it on both platforms and have less surface area for edge cases?

Reply:
I probably don't have an answer for that besides "that gives bad vibes". would you like me to change it?